tests/integration/s3/test_s3.py

```diff
@@ -4752,6 +4752,38 @@ def test_s3_intelligent_tier_config(self, aws_client, s3_create_bucket, snapshot
         response = aws_client.s3.list_bucket_intelligent_tiering_configurations(Bucket=bucket)
         snapshot.match("list_bucket_intelligent_tiering_configurations_3", response)
 
+    @pytest.mark.aws_validated
+    def test_s3_get_object_headers(self, aws_client, s3_create_bucket, snapshot):
+        bucket = s3_create_bucket()
+        key = "en-gb.wav"
+        file_path = os.path.join(f"../files/{key}")
+        aws_client.s3.upload_file(file_path, bucket, key)
+        objects = aws_client.s3.list_objects(Bucket=bucket)
+        etag = objects["Contents"][0]["ETag"]
+
+        try:
+            aws_client.s3.get_object(Bucket=bucket, Key=key, IfNoneMatch=etag)
+        except ClientError as e:
+            snapshot.match("if_none_match_err_1", e.response)
```

Comment:
nit: we can use `pytest.raises`, same for the one under
```suggestion
        with pytest.raises(ClientError) as e:
            aws_client.s3.get_object(Bucket=bucket, Key=key, IfNoneMatch=etag)
        snapshot.match("if_none_match_err_1", e.value.response)
```